pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Parser.java

```diff
@@ -48,19 +48,25 @@
         private final String filepath;
         private final String sourceText;
         private final SemanticErrorReporter reporter;
+        private final ClassLoader auxclasspathClassLoader;
 
         private final PropertySource propertySource;
 
-        public ParserTask(LanguageVersion lv, String filepath, String sourceText, SemanticErrorReporter reporter) {
+        public ParserTask(LanguageVersion lv, String filepath, String sourceText, SemanticErrorReporter reporter, ClassLoader auxclasspathClassLoader) {
             this.lv = Objects.requireNonNull(lv, "lv was null");
             this.filepath = Objects.requireNonNull(filepath, "filepath was null");
             this.sourceText = Objects.requireNonNull(sourceText, "sourceText was null");
             this.reporter = Objects.requireNonNull(reporter, "reporter was null");
+            this.auxclasspathClassLoader = Objects.requireNonNull(auxclasspathClassLoader);
 
             this.propertySource = new ParserTaskProperties();
             propertySource.definePropertyDescriptor(COMMENT_MARKER);
         }
 
+        public ParserTask(LanguageVersion lv, String filepath, String sourceText, SemanticErrorReporter reporter) {
+            this(lv, filepath, sourceText, reporter, Parser.class.getClassLoader());
```

Comment:
Should we really provide this default? I guess, the auxclasspath will be wrong all of the time in PMD7 that way...
Wait - I guess, this constructor is for languages, that don't need an auxclasspath, right?

Ah, I see, the auxclasspath will anyway be moved out of here into language properties. Good.